lib/pacio_test_kit/interactions_test.rb

```diff
@@ -60,12 +81,35 @@ def valid_json?(json)
       false
     end
 
+    def validate_response_metadata(response_resource, submitted_resource, method_type)
+      if method_type == 'create'
+        response = JSON.parse(response_resource)
+        submitted = JSON.parse(submitted_resource)
+
+        assert((response['id']).present?, 'Server SHALL populate the id for the newly created resource.')
+        assert(response['id'] != submitted['id'], 'Server response resource and submitted resource SHALL have different ids for the newly created resource.')
+
+        # if both blank, throw error; if both not blank, check if equal
+        if response['meta']['versionId'].blank? && submitted['meta']['versionId'].blank?
+          assert(!response['meta']['versionId'].blank? && !submitted['meta']['versionId'].blank?, 'Server response resource and submitted resource SHALL have different versionId fields for the newly created resource.')
+        elsif !response['meta']['versionId'].blank? && !submitted['meta']['versionId'].blank?
+          assert((response['meta']['versionId'] != submitted['meta']['versionId']), 'Server response resource and submitted resource SHALL have different versionId fields for the newly created resource.')
+        end
+
+        if response['meta']['lastUpdated'].blank? && submitted['meta']['lastUpdated'].blank?
+          assert(!response['meta']['lastUpdated'].blank? && !submitted['meta']['lastUpdated'].blank?, 'Server response resource and submitted resource SHALL have different lastUpdated fields for the newly created resource.')
+        elsif !response['meta']['lastUpdated'].blank? && !submitted['meta']['lastUpdated'].blank?
+          assert(response['meta']['lastUpdated'] != submitted['meta']['lastUpdated'], 'Server response resource and submitted resource SHALL have different lastUpdated fields for the newly created resource.')
+        end
+      end
+    end
```

Comment:
I think validate_meta_field is missing the case where both response and submitted resource fields = nil (or '')

Reply:
I omitted that check because the `meta` is not a required element. Plus, If you look at the [metadata description](https://build.fhir.org/resource.html#meta), you'll see that all the meta fields as well are not required since the have a cardinality of `(0..1)`.

The description says:
> The metadata items are all optional, though some or all of them may be required in particular implementations or contexts of use.

But none of the profiles define in the PFE IG has a `(1..1)` cardinality on the `meta` element.
So, it will be wrong to have that check &  fail a server if it does not populate the meta.